executor/prebackup.go

```diff
@@ -0,0 +1,142 @@
+		return true, nil
+	}
+
+	deployments := b.generateDeployments(templates.Items)
+
+	return b.startAllPreBackupDeployments(deployments)
+}
+
+// startAllPreBackupDeployments attempts to start all given deployment templates.
+// If some of them are already existing, their status will determine whether the backup is considered failed.
+// It returns true if all deployments are ready, also the ConditionPreBackupPodsReady will be set to ReasonReady.
+// Otherwise it returns false, indicating that not all PreBackup deployment(s) are ready.
+func (b *BackupExecutor) startAllPreBackupDeployments(deployments []appsv1.Deployment) (bool, error) {
+	existingDeployments := make([]*appsv1.Deployment, 0)
+	for _, deployment := range deployments {
+		exists, err := b.createPreBackupDeployment(deployment)
+		if err != nil {
+			return false, err
+		}
+		if exists {
+			deployment, err := b.fetchPreBackupDeployment(deployment.Name, deployment.Namespace)
+			if err != nil {
+				return false, err
+			}
+			existingDeployments = append(existingDeployments, deployment)
+		} else {
+			b.Log.Info("started pre backup pod", "namespace", deployment.Namespace, "name", deployment.Name)
+		}
+	}
+	if ready, err := b.allDeploymentsAreReady(existingDeployments); ready {
+		b.Log.Info("pre backup pod(s) now ready")
+		b.SetConditionTrue(k8upv1alpha1.ConditionPreBackupPodsReady, k8upv1alpha1.ReasonReady)
+		return true, nil
+	} else if err == nil {
+		b.SetConditionUnknownWithMessage(k8upv1alpha1.ConditionPreBackupPodsReady, k8upv1alpha1.ReasonWaiting, "waiting for %d PreBackupPods to become ready", len(deployments))
+	}
+	return false, nil
+}
+
+// allDeploymentsAreReady returns true if all given pre backup deployments have replicas in ready state.
+// If one of the deployment has failed (determined via "Progressing" condition), then it returns false with an error.
+// In that case, a backup is considered failed completely and status conditions will be set accordingly.
+// If one deployment is neither failed nor ready, then it returns false without error, indicating that it's still waiting.
+func (b *BackupExecutor) allDeploymentsAreReady(deployments []*appsv1.Deployment) (bool, error) {
+	readyCounter := 0
+	for _, deployment := range deployments {
+		ready, err := isPreBackupDeploymentReady(deployment)
+		log := b.Log.WithValues("preBackup", deployment.Name+"/"+deployment.Namespace)
+		if err != nil {
+			log.Info("backup failed: deadline exceeded on pre backup deployment")
+			b.SetConditionFalseWithMessage(k8upv1alpha1.ConditionPreBackupPodsReady, k8upv1alpha1.ReasonFailed, err.Error())
+			b.SetConditionTrueWithMessage(k8upv1alpha1.ConditionReady, k8upv1alpha1.ReasonFailed, err.Error())
+			b.deletePreBackupDeployment(deployment)
+			return false, err
+		}
+		if ready {
+			log.V(2).Info("pre backup pod is in ready state")
+			readyCounter++
+		} else {
+			log.Info("still waiting on pre backup pod...")
```

Comment:
```suggestion
			log.Info("still waiting on pre backup pod...")
```

Perhaps you could add here which pre backup pod we're currently wating for. It could make it easier to debug a scenario where such a pre backup pod is stuck.

Reply:
that is also set from `b.Log.WithValues("preBackup", deployment.Name+"/"+deployment.Namespace)` above. The return value of `WithValues` is a logger instance that has this field set in all following Info() calls